api/v1alpha3/azurecluster_default.go

```diff
@@ -34,10 +34,17 @@ func (c *AzureCluster) setDefaults() {
 }
 
 func (c *AzureCluster) setNetworkSpecDefaults() {
+	c.setResourceGroupDefault()
 	c.setVnetDefaults()
 	c.setSubnetDefaults()
 }
 
+func (c *AzureCluster) setResourceGroupDefault() {
```

Comment:
if we're going to default it, should also make the field optional? that would be a breaking change for v1alpha3 -> v1alpha2 conversion unfortunately

Reply:
IIRC, webhook defaulting and validating occur before API server schema validation. So we'd want the API server to see this field as required and the webhook to populate it if empty. I think this gets the behavior we want

Reply:
let me play around with some yaml to verify that

Reply:
thinking through this a bit -- switching to optional (and I assume a pointer) is typically useful when we want to differentiate between a type's zero value and that field not being sent at all. An example off the top of my head would be replicas -- sending a value of zero and not sending a value for a deployment spec result in different behavior: not sending a value gets defaulted inside the controller, sending 0 passes through as-is.

Empty string and no value would have the same behavior here. We always need this field defined from Azure perspective, so I think this is already semantically the most correct -- strictly requiring it, and simplifying the UX to autopopulate it.

Reply:
Checked out your branch and removed the "resourceGroup" field, I'm getting `error: error validating "STDIN": error validating data: ValidationError(AzureCluster.spec): missing required field "resourceGroup" in io.x-k8s.cluster.infrastructure.v1alpha3.AzureCluster.spec; if you choose to ignore these errors, turn validation off with --validate=false`

Reply:
I think there's a difference between `resourceGroup: ""` and no resourceGroup specified at all

Reply:
we can mark it as `// +optional` but not make it a pointer, just like vnetSpec

Reply:
sad, using `--v=10` it looks like kubectl uses discovery to do client-side schema validation, even though it would work with the webhooks properly.

will do as you suggested